daemon/src/identify.rs

```diff
@@ -0,0 +1,182 @@
+use crate::Environment;
+use std::collections::HashSet;
+
+pub mod dialer;
+pub mod listener;
+pub mod protocol;
+
+pub const PROTOCOL: &str = "/itchysats/id/1.0.0";
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PeerInfo {
+    pub wire_version: String,
+    pub daemon_version: String,
+    pub environment: Environment,
+    pub protocols: HashSet<String>,
+}
+
+impl TryFrom<protocol::IdentifyMsg> for PeerInfo {
+    type Error = ConversionError;
+
+    fn try_from(identity_msg: protocol::IdentifyMsg) -> Result<Self, Self::Error> {
+        let identity_info = PeerInfo {
+            wire_version: identity_msg.wire_version(),
+            daemon_version: identity_msg
+                .daemon_version()
+                .map_err(|error| ConversionError { error })?,
+            environment: identity_msg.environment().into(),
+            protocols: identity_msg.protocols(),
+        };
+
+        Ok(identity_info)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Conversion to identity info failed: {error}")]
+pub struct ConversionError {
+    #[source]
```

Comment:
🙃 And then you don't have to map explicitly on line 26.

```suggestion
    #[from]
```